vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -303,6 +304,24 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+// CompareAndSwap ep's serviceEnabled. If its in oldState, set it to newState
+// and return true.  If its not in oldState return false
+func (ep *endpoint) casServiceEnabled(oldState, newState bool) bool {
```

Comment:
This is kind of a bazaar API. I'm not sure that CAS works unless you know the last observed revision, not just value. For example, below, the value is not actually observed. It seems like you just need to know if the action succeeded or not.